### PR TITLE
Display average course rating in the search result.

### PR DESCRIPTION
**Pull request by ZiqiXiong**

### CODE REVIEW

File: aspc/courses/templatetags/stars.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+from django.utils.safestring import mark_safe
+
+register = template.Library()
+
+@register.filter
+def star(rating):
+    num_star_on = int(math.floor(rating))
+    num_star_half = int(round(rating)) - num_star_on
+    num_star_off = 5 - num_star_on - num_star_half
+    html = num_star_on * '<img src="/static/images/star-on.gif">' + \
```
Comment (mattdahl) [inline aspc/courses/templatetags/stars.py]: I did not know you could do string multiplication in Python like this - that's awesome

Comment (mattdahl) [inline aspc/courses/templatetags/stars.py]: can you do `from aspc import settings` and reference `settings.STATIC_URL` as the beginning of the `src` attribute for these nodes? so the url isn't hard-coded

File: aspc/courses/models.py [hunk 2]
```diff
@@ -158,7 +158,9 @@ def json(self):
         return {'events': event_list, 'info': {'course_code': self.code, 'course_code_slug': self.code_slug,
                                                'detail_url': self.get_absolute_url(),
                                                'campus_code': self.get_campus(), }}
-
+    def get_average_rating(self):
```
Comment (mattdahl) [inline aspc/courses/models.py]: I think you are correct to have this method here - does this mean that we can get rid of the `CourseReview#get_average_course_instructor_rating` declaration below (line 307)? These two methods do the same thing - when would we want to be able to call that from a `CourseReview` object instead? @alokedesai

Comment (alokedesai) [inline aspc/courses/models.py]: @mattdahl @ZiqiXiong Yep, I agree that this seems like the more logical place to put it.

File: aspc/courses/templates/courses/course_list_item.html [hunk 3]
```diff
@@ -1,5 +1,10 @@
+{% load stars %}
 <h3 {% if ctx == "schedule" %}id="course_{{ c.code_slug }}" data-code="{{ c.code_slug }}" data-added="{{ c.added }}"{% endif %}><a href="{{ c.get_absolute_url }}">{{ c.code }} &mdash; {{ c.course.name }}</a></h3>
-<h4>{{ c.instructors.all|join:", " }} &mdash; credit: {{ c.credit }} <!-- &mdash; {% if c.filled %}{{ c.filled }} filled of {% endif %}{{ c.spots }} seats --></h4>
+<h4>{{ c.instructors.all|join:", " }} &mdash; credit: {{ c.credit }}
+    {% if c.get_average_rating %}
```
Comment (mattdahl) [inline aspc/courses/templates/courses/course_list_item.html]: Can you group all the stars into a `<div>`? I also think that that `<div>` should go after the `<h4>` tag here, not within it, so the stars display on a new line below the instructor name. And I think we should display some text, i.e. "Average rating:", in front the stars too.

Also, right now nothing displays if there are no reviews for a course. I think instead we ought to show some text, e.g. "No reviews yet. Add your own!", in order to encourage people to submit reviews for such courses.

File: aspc/courses/templatetags/__init__.py [hunk 4]
```diff
@@ -0,0 +1 @@
+__author__ = 'zxiong'
```
Comment (alokedesai) [inline aspc/courses/templatetags/__init__.py]: @ZiqiXiong Any reason you didn't just make the init py blank? I don't think we add the author like this anywhere in the codebase? You can always git blame to look for who the author is

**Comment by ZiqiXiong** [comment]
@mattdahl Thanks for the suggestions. I will make the changes. @alokedesai It was automatically added by the IDE I used. I will remove it!

**Comment by mattdahl** [comment]
Also, did you author these commits from within the VM? (It says the `vagrant` user did.) If it's not too much trouble, can you re-author them as yourself instead so the commit history preserves a useful record of who did what?

**Comment by ZiqiXiong** [comment]
@mattdahl Not a problem.

**Pull request merged**